docs/plots/backends.rst

```diff
@@ -0,0 +1,384 @@
+************************
+Plotting backend details
+************************
+
+.. warning::
+   The plotting backend API is currently re-designed and details might change in the near future.
```

Comment:
```suggestion
   The plotting backend API should be considered experimental for 4.16.0.
```

Reply:
I'm working through those comments and "resolve conversation" for those that I've addressed locally. In same cases, I'm not taking the exact wording suggested in review (e.g. here I prefer "currently" instead of "4.16.0" just in case we will consider it experimental in 4.16.1 so we don't have to update that comment every time.)